### PR TITLE
CUMULUS-2089 Granule Not Found Table

### PR DESCRIPTION
**Pull request by dopeters**
**Summary:** Biggest change here is breaking out the `ReconciliationReport` component to display either `InventoryReport` or `GnfReport` based on the type. 

Addresses [CUMULUS-2089: Reports: GNF View - Table Pattern](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2089)

## Changes

* Created separate components for each report type
* Added table config for GNF reports
* Made download functions reusable and updated them so that we don't have to disable lint anymore
* Calculate GNF indicators based on Lauren's comment [here](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1829?focusedCommentId=301857&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-301857)

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: app/src/js/components/ReconciliationReports/reconciliation-report.js [hunk 1]
```diff
@@ -1,87 +1,23 @@
...
-    reportComparisons.reduce((object, item) => {
-      object[item.id] = item.tables.reduce((tableObject, table, index) => {
-        // expand the zeroth table in each bucket. collapse all others
-        tableObject[table.id] = index === 0;
-        return tableObject;
-      }, {});
-      return object;
-    }, {})
-  );
+  const { type: reportType = 'Inventory' } = recordData || {};
```
Comment (dopeters) [inline app/src/js/components/ReconciliationReports/reconciliation-report.js:19]: Currently, to test the GNF view, you can change this default value to 'gnf'.

**Comment by bt6453** [comment]
Checked the GNF example and everything looks good!

File: app/src/js/components/ReconciliationReports/gnf-report.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  const { onlyInCmr = [], onlyInCumulus = [] } = granulesInCumulusCmr;
+
+  const cmrGranules = onlyInCmr.map((granule) => {
+    const { GranuleUR } = granule;
+    return {
+      granuleId: GranuleUR,
+      cmr: true,
+      cumulus: false,
+      s3: false,
+    };
```
Comment (jennyhliu) [inline app/src/js/components/ReconciliationReports/gnf-report.js:42]: We could get collectionId here, construct it from `shortName` and `Version`.

File: test/components/reconciliation-reports/reconciliation-report.js [hunk 3]
```diff
@@ -11,12 +11,50 @@ configure({ adapter: new Adapter() });
...
+        },
+        granulesInCumulusCmr: {
+          okCount: 7
+        },
+        filesInCumulusCmr: {
+          okCount: 4
+        }
+      }
+    },
+    // TODO: change this once we have the actual GNF format
```
Comment (jennyhliu) [inline test/components/reconciliation-reports/reconciliation-report.js:50]: remove the TODO?

File: cypress/integration/reconciliation_spec.js [hunk 4]
```diff
@@ -69,24 +69,25 @@ describe('Dashboard Reconciliation Reports Page', () => {
...
       cy.visit('/reconciliation-reports');
       cy.get('[data-value="inventoryReport-20200114T202529026"]').find('.button__row--delete').click({ force: true });
 
-      cy.get('.table .tbody .tr').should('have.length', 2);
+      cy.get('.table .tbody .tr').should('have.length', 3);
       cy.get('[data-value="inventoryReport-20200114T202529026"]')
         .should('not.exist');
     });
 
     it('displays a link to an individual report', () => {
```
Comment (jennyhliu) [inline cypress/integration/reconciliation_spec.js:84]: Do you have a test for GNF report page?

File: package.json [hunk 5]
```diff
@@ -67,7 +67,7 @@
     "@babel/preset-react": "^7.0.0",
     "@babel/register": "^7.0.0",
     "@babel/runtime": "^7.8.4",
-    "@cumulus/api": "2.0.1",
+    "@cumulus/api": "^2.0.3-alpha.0",
```
[the remark this answers is not in the document]
Comment (dopeters) [reply package.json]: Done https://github.com/nasa/cumulus-dashboard/pull/840/commits/374cc6cc2debfd8bc8d4aef0d02f01a50c6ca384

File: app/src/js/components/ReconciliationReports/gnf-report.js (hunk 2 above)
Comment (dopeters) [reply app/src/js/components/ReconciliationReports/gnf-report.js:45]: https://github.com/nasa/cumulus-dashboard/pull/840/commits/0cb97ec11a56d4928ab1176f9212dd7fe61b2b00

File: test/components/reconciliation-reports/reconciliation-report.js (hunk 3 above)
Comment (dopeters) [reply test/components/reconciliation-reports/reconciliation-report.js]: Nice catch! https://github.com/nasa/cumulus-dashboard/pull/840/commits/63672daedb9427b2e75f04f50608d770fc3aff72

File: cypress/integration/reconciliation_spec.js (hunk 4 above)
Comment (dopeters) [reply cypress/integration/reconciliation_spec.js]: Added one https://github.com/nasa/cumulus-dashboard/pull/840/commits/7d2937ba468683ec9d3dc6805f3910580f2c57f0

**Review by jennyhliu — APPROVED** [review]
Nice job 👍 Tested with the GNF reports generated from my stack, and everything looks good.

**Pull request merged**